### DISCUSSION TITLE
[Feature request] Allow to ignore certain files when running deploy/revert/verify

### DISCUSSION THREAD

**Discussion by andreihojda** [theory/sqitch#761] [General]
I am having the following use-case: I have sql files containing both DDL ( tables and functions manily ) and queries. After running the deploy I am running some regression and behavioral tests on the functions on an empty database. The issue i'm facing is that I can't ignore the query files from sqitch.plan and this causes tests to be inconsistent. 
Would it make sense to have maybe a flag to allow to ignore a certain pattern of sql files when running deploy/revert/verify. Something like:
`sqitch deploy --ignore QUERY* db:pg://user:pass@server:port/database`

**Comment by theory** [comment]
Sqitch expects the files to have the same name as changes but ending in `.sql`. So if you don't use the change names for your test files it should work.

Personally, I like to keep test files in a different directory, it keeps things more compartmentalized.

**Comment by andreihojda** [comment]
Maybe i didn't explain my use case well enough. I am writing DDL and DML in different files in the same project.
An example would be a continent table and populating it with the continents. This would leave me with 2 `.sql` files. 
I would like to be able to run only DDL or DML `.sql` files when deploying something. When i run my tests, i want to run only DDL, when i deploy everything, i want to run both DDL and DML. 

I was thinking that filtering which files to include in the deploy would allow me to do that.

**Comment by theory** [comment]
In that case, I think you might want two plans. The [Separate Plans section of the sqitch-configuration guide](http://sqitch.org/docs/manual/sqitch-configuration/#separate-plans) describes how, in this case to support two engines, but in your case, you'd have one plan and set of deploy/revert/verify files for DDL and another set for DML, living in separate directories.

What you're asking for will never work, because Sqitch uses a merkle chain to ensure that changes are deployed in the proper order. I you omit any from the plan, it won't work at all.
